apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -112,132 +197,259 @@ private static CodeableConcept getPriority() {
-          new Claim.ProcedureComponent()
-              .setSequence((i + 1))
-              .setDate(
-                  procCode.getProcDate() == null
-                      ? null
-                      : Date.from(
-                          procCode.getProcDate().atStartOfDay(ZoneId.systemDefault()).toInstant()))
-              .setProcedure(
-                  new CodeableConcept()
-                      .setCoding(
-                          Collections.singletonList(
-                              new Coding(
-                                  "http://hl7.org/fhir/sid/icd-10",
-                                  procCode.getProcCode(),
-                                  null))));
-
-      procedure.add(component);
+                    new CodeableConcept(
+                        new Coding(
+                            C4BBIdentifierType.NPI.getSystem(),
+                            C4BBIdentifierType.NPI.toCode(),
+                            C4BBIdentifierType.NPI.getDisplay())))
+                .setSystem(TransformerConstants.CODING_NPI_US)
+                .setValue(claimGroup.getNpiNumber()));
+      }
+
+      if (claimGroup.getLobCd() != null) {
+        reference.setExtension(
+            List.of(
+                new Extension(BBCodingSystems.CLM_FACILITY_TYPE_CODE)
+                    .setValue(
+                        new Coding(
+                            BBCodingSystems.CLM_FACILITY_TYPE_CODE, claimGroup.getLobCd(), null))));
+      }
+
+      return reference;
     }
 
-    return procedure;
+    return null;
+  }
+
+  private static List<Claim.DiagnosisComponent> getDiagnosis(
+      PreAdjFissClaim claimGroup, boolean isIcd9) {
+    final String systemSuffix = isIcd9 ? "9-cm" : "10";
+
+    return ObjectUtils.defaultIfNull(claimGroup.getDiagCodes(), List.<PreAdjFissDiagnosisCode>of())
+        .stream()
+        .map(
+            diagnosisCode -> {
+              Claim.DiagnosisComponent component =
+                  new Claim.DiagnosisComponent()
+                      .setSequence(diagnosisCode.getPriority())
+                      .setDiagnosis(
+                          new CodeableConcept()
+                              .setCoding(
+                                  Collections.singletonList(
+                                      new Coding(
+                                          "http://hl7.org/fhir/sid/icd-" + systemSuffix,
+                                          diagnosisCode.getDiagCd2(),
+                                          null))));
+
+              if (diagnosisCode.getDiagPoaInd() != null // Present on Admission
+                  && WHITE_LISTED_POA.contains(diagnosisCode.getDiagPoaInd().toLowerCase())) {
+                component.setOnAdmission(
+                    new CodeableConcept(
+                        // If any other value, omit
+                        new Coding(
+                            "http://terminology.hl7.org/CodeSystem/ex-diagnosis-on-admission",
```

Comment:
I'm still weighing this with using the EOB coding system, https://bluebutton.cms.gov/resources/variables/clm_poa_ind_sw1. No changes yet, but this is something I'll want to discuss with the BFD team